app/scripts/modules/core/src/pipeline/config/triggers/triggers.directive.js

```diff
@@ -38,6 +38,15 @@ module.exports = angular.module('spinnaker.core.pipeline.config.trigger.triggers
       kind: 'custom'
     });
 
+    this.removeExpectedArtifact = (pipeline, expectedArtifact) => {
```

Comment:
can we safely assume the existence of the `pipeline.expectedArtifacts` and `pipeline.triggers` arrays in this func?

Reply:
Yup, the only way to run this is from a button that appears when an ng-repeat creates at least one element from the expected artifact list. May as well be thorough -- will add extra checks.